src/gui/dialogs/title_screen.cpp

```diff
@@ -315,8 +315,22 @@ void title_screen::init_callbacks()
 	//
 	// Preferences
 	//
-	register_button(*this, "preferences", hotkey::HOTKEY_PREFERENCES, []() {
+	register_button(*this, "preferences", hotkey::HOTKEY_PREFERENCES, [this]() {
 		gui2::dialogs::preferences_dialog::display();
+		// The screen may have changed size, so to be safe,
+		// ensure that blurred elements recalculate their blur.
+		// Better would be to update every styled_widget,
+		// but for now this just grabs the two we know have blur.
+		panel* tip_panel = find_widget<panel>(this, "tip_panel", false, false);
```

Comment:
The (wesnoth) window can also be resized without the preferences dialog (by dragging the window border with the mouse like any window) in that case this code is not called. So when the case that it changed through the preferences dialog needs special handling here, it's be useful to have a comment why that is.

Reply:
You're right it's not very clear. I'll update the description there a bit

Reply:
I made the description clearer about what it's doing in this case. If it turns out to be an issue that this doesn't cover resizing via dragging the window borders, this tweak may have to be adapted or moved elsewhere.